ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/StatementExecutor.java

```diff
@@ -386,4 +404,27 @@ private void terminateQuery(final TerminateQuery terminateQuery) {
           String.format("No running query with id %s was found", queryId));
     }
   }
+
+  private void throwTooManyActivePersistentQueriesException(final String statementStr) {
+    throw new KsqlException(
+        String.format(
+            "Not executing statement '%s' since the statement causes the limit on number "
+                + "of active, persistent queries to be exceeded "
+                + "(%d persistent queries currently running. Limit is %d). "
+                + "Use the TERMINATE command to terminate existing queries, "
+                + "or reconfigure the limit via the 'ksql-server.properties' file.",
+            statementStr,
+            ksqlEngine.numberOfPersistentQueries(),
+            ksqlConfig.getInt(KsqlConfig.KSQL_ACTIVE_PERSISTENT_QUERY_LIMIT_CONFIG)
+        )
+    );
+  }
+
+  private int getPersistentQueryLimit(final Command command) {
+    return Integer.parseInt(
+        command.getOriginalProperties().getOrDefault(
```

Comment:
Instead of getting the config from the command here, you should pass in a ksqlConfig. The ksqlConfig should be created by calling overrideBreakingConfigsWithOriginalValues. So from the caller, do:

```
final KsqlConfig mergedConfig = ksqlConfig.overrideBreakingConfigsWithOriginalValues(command.getOriginalProperties());
if (getPersistentQueryLimit(command, mergedConfig)...
```

And in this function:
```
ksqlConfig.getInt(KsqlConfig. KSQL_ACTIVE_PERSISTENT_QUERY_LIMIT_CONFIG)
```

This way, KsqlConfig takes care of using the right value or default for you.